torch_xla/csrc/tensor_methods.cpp

```diff
@@ -896,7 +896,11 @@ XLATensorPtr clamp(const XLATensorPtr& input,
 }
 
 XLATensorPtr clone(const XLATensorPtr& input) {
-  return input->CreateFrom(input->GetIrValue());
+  XLATensorPtr cloned = input->CreateFrom(input->GetIrValue());
+  if (input->sharding_spec() != nullptr) {
+    cloned->SetShardingSpec(*input->sharding_spec());
```

Comment:
maybe pass `sharding_spec` to `CreateFrom` directly?

Reply:
I'd prefer them separate, so that each API has smaller and more specific scope, thus more modular. If need to, let's address the API changes in a separate PR.

Reply:
ok after I did a quick grep, `CreateFrom` was used in a lot of different methods under `tensor_methods` we for the most of them we don't want to inherit the `sharding_spec`, separting them actually is the right thing to do here.